netplugin/netd.go

```diff
@@ -75,238 +49,123 @@ func startNetPlugin(pluginConfig *plugin.Config) {
-			Usage:  "set netplugin log format to json",
-		},
-		cli.BoolFlag{
-			Name:   "use-syslog, syslog",
-			EnvVar: "CONTIV_NETPLUGIN_USE_SYSLOG",
-			Usage:  "set netplugin send log to syslog or not",
-		},
-		cli.StringFlag{
-			Name:   "syslog-url",
-			Value:  "udp://127.0.0.1:514",
-			EnvVar: "CONTIV_NETPLUGIN_SYSLOG_URL",
-			Usage:  "set netplugin syslog url in format protocol://ip:port",
-		},
+	netpluginFlags := []cli.Flag{
 		cli.StringFlag{
 			Name:   "host, host-label",
-			Value:  hostname,
 			EnvVar: "CONTIV_NETPLUGIN_HOST",
-			Usage:  "set netplugin host to identify itself",
-		},
-		cli.StringFlag{
-			Name:   "mode, plugin-mode",
-			EnvVar: "CONTIV_NETPLUGIN_MODE",
-			Usage:  "set netplugin mode, options: [docker, kubernetes, swarm-mode]",
+			Usage:  "set netplugin host to identify itself (default: <host-name-reported-by-the-kernel>)",
 		},
 		cli.StringFlag{
 			Name:   "vtep-ip",
-			Value:  localIP,
 			EnvVar: "CONTIV_NETPLUGIN_VTEP_IP",
-			Usage:  "set netplugin vtep ip for vxlan communication",
+			Usage:  "set netplugin vtep ip for vxlan communication (default: <host-ip-from-local-resolver>)",
 		},
 		cli.StringFlag{
 			Name:   "ctrl-ip",
-			Value:  localIP,
 			EnvVar: "CONTIV_NETPLUGIN_CONTROL_IP",
-			Usage:  "set netplugin control ip for control plane communication",
+			Usage:  "set netplugin control ip for control plane communication (default: <host-ip-from-local-resolver>)",
 		},
 		cli.StringFlag{
 			Name:   "vlan-uplinks, vlan-if",
 			EnvVar: "CONTIV_NETPLUGIN_VLAN_UPLINKS",
 			Usage:  "a comma-delimited list of netplugin uplink interfaces",
 		},
-		cli.StringFlag{
-			Name:   "etcd-endpoints, etcd",
-			EnvVar: "CONTIV_NETPLUGIN_ETCD_ENDPOINTS",
-			Usage:  "a comma-delimited list of netplugin etcd endpoints",
-		},
-		cli.StringFlag{
-			Name:   "consul-endpoints, consul",
-			EnvVar: "CONTIV_NETPLUGIN_CONSUL_ENDPOINTS",
-			Usage:  "a comma-delimited list of netplugin consul endpoints, ignored when etcd-endpoints is set",
-		},
 		cli.IntFlag{
 			Name:   "vxlan-port",
 			Value:  4789,
 			EnvVar: "CONTIV_NETPLUGIN_VXLAN_PORT",
 			Usage:  "set netplugin VXLAN port",
 		},
-		cli.StringFlag{
-			Name:   "netmode, network-mode",
-			EnvVar: "CONTIV_NETPLUGIN_NET_MODE",
-			Usage:  "set netplugin network mode, options: [vlan, vxlan]",
-		},
-		cli.StringFlag{
-			Name:   "fwdmode, forward-mode",
-			EnvVar: "CONTIV_NETPLUGIN_FORWARD_MODE",
-			Usage:  "set netplugin forwarding network mode, options: [bridge, routing]",
-		},
-		/*
-			// only ovs is supported
-			// TODO: turn it on when having more than one backend supported
-			cli.StringFlag {
-				Name: "driver, net-driver",
-				Value: "ovs",
-				EnvVar: "CONTIV_NETPLUGIN_DRIVER",
-				Usage: "set netplugin key-value store url, options: [ovs, vpp]",
-			}
-		*/
 	}
+	app.Flags = utils.FlattenFlags(netpluginFlags, utils.BuildDBFlags(binName), utils.BuildNetworkFlags(binName), utils.BuildLogFlags(binName))
```

Comment:
depends on how many appends you want to put into codes. it will be like 
`app.Flags = append(append(append(netpluginFlags, utils.BuildDBFlags(binName)...), utils.BuildNetworkFlags(binName)...), utils.BuildLogFlags(binName)...)`
I think it's very ugly